### PR TITLE
Mockgen

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/solo-kit/issues/150
resolves https://github.com/solo-io/solo-kit/issues/160

### CODE REVIEW

**Comment by EItanya** [comment]
> very nice. 2 comments:
> 
>     * not sure there's any value to mocking out the `Cloneable X Interface` as its only reason for existence is to make it easier to run the `Implement Interface...` action from the IDE. it's not implemented anywhere
> 
>     * it'd be nice to see a test in solokit that makes use of some of this new generated code

@ilackarms about the `Cloneable` interface, I will see if there is an easy way to skip specific resources, but the program just grabs all interfaces from the file, and skipping specific ones may be more trouble than it is worth.
As far as tests, I'll write a few

File: pkg/code-generator/cmd/main.go [hunk 1]
```diff
@@ -26,6 +28,10 @@ import (
 
 type DocsOptions = options.DocsOptions
 
+const (
+	TestMode = "TEST_MODE"
```
Comment (ilackarms) [inline pkg/code-generator/cmd/main.go]: unclear name
`SKIP_MOCK_GEN`?

File: ci/check-code-gen.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ PATH=/workspace/gopath/bin:$PATH
 
 set +e
 
-make generated-code -B  > /dev/null
+make generated-code -B
```
Comment (EItanya) [inline ci/check-code-gen.sh]: This is useful output because it's solo-kit, so seeing errors here can be informative

**Pull request merged**